docs/reference/settings/security-settings.asciidoc

```diff
@@ -1858,6 +1858,21 @@ connections allowed per endpoint.
 Defaults to `200`.
 // end::oidc-http-max-endpoint-connections-tag[]
 
+// tag::oidc-http-connection-pool-ttl-tag[]
+`http.connection_pool_ttl` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the OpenID Connect Provider endpoints. Specifies the time-to-live of connections
+in the connection pool. If a connection is not re-used within this timeout,
+it is closed.
+By default, the time-to-live is 3 minutes meaning that connections are closed
+after being idle for 3 minutes.
+
+Configure this setting to `-1` to let the server dictates this value using the `Keep-Alive` HTTP
+response header. If the header is not set by the server, the time-to-live is infinite meaning
+that connections never expire.
```

Comment:
I think we should reword this, but I'll come back with a suggestion later.

Reply:
I reword'd this paragraph and also made it consistent to the new behaviour of handling keep-alive response header.